### PR TITLE
Creating custom supply chains in TAP

### PR DESCRIPTION
**Pull request by dkhopade**
In this workshop content, Gary Gutman leveraged the content from Timo Salm's repo, however, we focused on explaining the steps to customize every component of TAP's OOTB testing. & scanning supply chains.
We have also provided a way to understand many small things, as well as choices to consider your build step with either Kaniko or kpack. 

This workshop content will start with a basic supply chain and then let the attendee extend the same basic supply chain into a more enterprise-ready supply chain with testing & scanning abilities.

### CODE REVIEW

File: workshops/create-custom-supply-chains.yaml [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+  difficulty: intermediate
+  #! The duration field gives the maximum amount of time the workshop takes to complete. 
+  #! This field provides informational value and does not guarantee how long a workshop instance lasts. 
+  #! The field format is an integer number with s, m, or h suffix.
+  duration: 3h
+  #! url: https://github.com/mrgaryg/tap-cartographer-workshop
+  #! url: #@ data.values.workshop_git.git_url + "/" + data.values.workshop_git.git_org_name + "/" + data.values.workshop_git.git_repo
+  url: https://github.com/dkhopade/tap-cartographer-workshop
+  content:
+    image: #@ "{}/tap-workshop/cartographer-workshop:latest".format(data.values.container_registry.hostname)
```
Comment (cdelashmutt-pivotal) [inline workshops/create-custom-supply-chains.yaml]: We don't have this data value in the waas-catalog config.  You could use Git's registry to store your workshop images, and then refer to them here.

File: workshops/create-custom-supply-chains.yaml [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+    files: #@ "{}/{}/tap-cartographer-workshop/workshop?ref=se-supplychains-workshop-dev".format(data.values.workshop_git.git_url,data.values.workshop_git.git_org_name)
```
Comment (cdelashmutt-pivotal) [inline workshops/create-custom-supply-chains.yaml]: We don't have `workshop_git` in the values for the WaaS catalog.  You'd probably want to point this to your source repo for the workshop content you want to pull in (github.com, for example).  Check out some of the other workshop definitions to see some examples.

File: workshops/create-custom-supply-chains.yaml [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+    - name: CONTAINER_REGISTRY_HOSTNAME
+      value: #@ data.values.container_registry.hostname
+    - name: CONTAINER_REGISTRY_USERNAME
+      value: #@ data.values.container_registry.username
+    - name: CONTAINER_REGISTRY_PASSWORD
+      value: #@ data.values.container_registry.password
+    - name: CONTAINER_REGISTRY_REPOSITORY
+      value: #@ data.values.container_registry.repository
+    - name: GITOPS_REPOSITORY
+      value: #@ "{}/$(session_namespace).git".format(data.values.gitops.https_repository_prefix)
```
Comment (cdelashmutt-pivotal) [inline workshops/create-custom-supply-chains.yaml]: We have a gitea install and user you can use at gitea.$(ingress_domain)/gitea_admin.  That server is configured to allow "Push-to-create" for repos, so you can create a git repo, and push to a non-existent repo and Gitea will create it.

File: workshops/create-custom-supply-chains.yaml [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+    - name: GITOPS_REPOSITORY_USERNAME
```
Comment (cdelashmutt-pivotal) [inline workshops/create-custom-supply-chains.yaml:65]: The user name for gitea is "gitea_admin" and password for "gitea_admin" can be injected by referencing `data.values.gitea.adminPassword`

File: workshops/create-custom-supply-chains.yaml [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
+              max:
+                storage: 10Gi
+      #! Permissions for tap-sa (Supply Chain)
+      - apiVersion: v1
+        kind: Secret
+        metadata:
+          name: tanzu-net-credentials
+        type: kubernetes.io/dockerconfigjson
+        data:
+          #@ docker_auth = base64.encode("{}:{}".format(data.values.tanzunet.username, data.values.tanzunet.password))
```
Comment (cdelashmutt-pivotal) [inline workshops/create-custom-supply-chains.yaml]: We don't have a Tanzunet username and password available via the WaaS catalog currently, so we'd need to add that.

**Review by cdelashmutt-pivotal — CHANGES_REQUESTED** [review]
There are a few items that would need to be addressed before we could merge this PR.  Have a look and let's discuss.

File: workshops/create-custom-supply-chains.yaml (hunk 5 above)
Comment (dkhopade) [reply workshops/create-custom-supply-chains.yaml]: So we found that we need this when we build an image; we can think of taking it off in future once we have a way within WaaS environment

File: workshops/create-custom-supply-chains.yaml (hunk 4 above)
Comment (dkhopade) [reply workshops/create-custom-supply-chains.yaml:65]: for now, using our GitHub public repo just to test and will come back with gitea that is hosted on WaaS

File: workshops/create-custom-supply-chains.yaml (hunk 3 above)
Comment (dkhopade) [reply workshops/create-custom-supply-chains.yaml]: for now, using our GitHub public repo just to test and will come back with gitea that is hosted on WaaS

File: workshops/create-custom-supply-chains.yaml (hunk 2 above)
Comment (dkhopade) [reply workshops/create-custom-supply-chains.yaml]: Taken care 👍

File: workshops/create-custom-supply-chains.yaml (hunk 1 above)
Comment (dkhopade) [reply workshops/create-custom-supply-chains.yaml]: Have removed ytt for now

**Comment by dkhopade** [comment]
Addressed critical review comments and issues for this 1st iteration. Thank you !!

**Pull request merged**